packages/twenty-front/src/modules/workflow/components/WorkflowEditActionFormServerlessFunction.tsx

```diff
@@ -23,6 +28,53 @@ export const WorkflowEditActionFormServerlessFunction = (
 
   const { serverlessFunctions } = useGetManyServerlessFunctions();
 
+  const form = useForm({
+    disabled: props.readonly,
+  });
+
+  const defaultFunctionInput = Object.fromEntries(
+    Object.entries(props.action.settings.input).filter(
+      ([key, _]) =>
+        key !== 'serverlessFunctionId' && key !== 'serverlessFunctionVersion',
+    ),
+  );
+
+  useEffect(() => {
+    form.reset();
+    Object.entries(props.action.settings.input).forEach(([key, value]) => {
+      form.setValue(key, value ?? '');
+    });
+  }, [props.action.settings.input, form]);
+
```

Comment:
We want to avoid useEffect in components, can you move that in a hook?
This hook would be responsible for providing a state initiated with `action.setttings.input`, a setState. We could use a useEffect to call `updateFunctionInput` when state is updated